io/zenoh-transport/Cargo.toml

```diff
@@ -77,5 +78,5 @@ zenoh-util = { workspace = true }
 env_logger = { workspace = true }
 panic-message = { workspace = true }
 zenoh-protocol = { workspace = true, features = ["test"] }
-zenoh-transport = { workspace = true, features = ["test"] }
+zenoh-transport = { workspace = true, features = ["test", "transport_multilink"] }
```

Comment:
What's the purpose of `"transport_multilink"` feature here?

Reply:
So, briefly, the tests for zenoh-transport do not compile without  `transport_multilink` feature!

When we do `cargo test`, `transport_multilink` feature is enabled by default in zenoh, so the tests compile.

But when we compile tests for separate test package, for example as we do on the CI, the compilation fails:
`cargo nextest run -F shared-memory -F transport_shm -p zenoh-transport`.

I think the proper way would be adding a support of `transport_multilink == false` for `zenoh-transport` tests, but this change actually adds conditional compilation to test scenarios and require testing for both modes. A treat this as a part of another workpackage (test refactoring and feature testing)

Reply:
Ok, this is then linked to the changes introduced in the CI. We are introducing the additional feature in `[dev-dependencies]` for test purposes. 

If I run the command provided by you (the same as in the CI) I get:
```
$ cargo nextest run -F shared-memory -F transport_shm -p zenoh-transport
error: Found argument '-F' which wasn't expected, or isn't valid in this context

	If you tried to supply `-F` as a value rather than a flag, use `-- -F`

USAGE:
    cargo nextest run [OPTIONS] [FILTERS]...

For more information try --help
```

Then, by following cargo's suggestion no tests are run at all:

```
$ cargo nextest run -- -F shared-memory -F transport_shm -p zenoh-transport
    Blocking waiting for file lock on build directory
   Compiling zenoh-protocol v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/commons/zenoh-protocol)
   Compiling zenoh-core v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/commons/zenoh-core)
   Compiling zenoh-sync v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/commons/zenoh-sync)
   Compiling zenoh-util v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/commons/zenoh-util)
   Compiling zenoh-codec v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/commons/zenoh-codec)
   Compiling zenoh-link-commons v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/io/zenoh-link-commons)
   Compiling zenoh-config v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/commons/zenoh-config)
   Compiling zenoh-plugin-trait v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/plugins/zenoh-plugin-trait)
   Compiling zenoh-link-ws v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/io/zenoh-links/zenoh-link-ws)
   Compiling zenoh-link-tcp v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/io/zenoh-links/zenoh-link-tcp)
   Compiling zenoh-link-udp v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/io/zenoh-links/zenoh-link-udp)
   Compiling zenoh-link-unixsock_stream v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/io/zenoh-links/zenoh-link-unixsock_stream)
   Compiling zenoh-link-quic v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/io/zenoh-links/zenoh-link-quic)
   Compiling zenoh-link-tls v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/io/zenoh-links/zenoh-link-tls)
   Compiling zenoh-link-shm v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/io/zenoh-links/zenoh-link-shm)
   Compiling zenoh-link-serial v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/io/zenoh-links/zenoh-link-serial)
   Compiling zenoh-link v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/io/zenoh-link)
   Compiling zenoh-transport v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/io/zenoh-transport)
   Compiling zenoh v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/zenoh)
   Compiling zenoh-ext v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/zenoh-ext)
   Compiling zenoh_backend_traits v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/plugins/zenoh-backend-traits)
   Compiling zenoh-examples v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/examples)
   Compiling zenoh-plugin-storage-manager v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/plugins/zenoh-plugin-storage-manager)
   Compiling zenoh-plugin-rest v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/plugins/zenoh-plugin-rest)
   Compiling zenoh-plugin-example v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/plugins/example-plugin)
   Compiling zenohd v0.10.0-dev (/Users/luca/Repository/eclipse-zenoh/zenohd)
    Finished test [unoptimized + debuginfo] target(s) in 1m 59s
    Starting 0 tests across 54 binaries (199 skipped)
     Summary [   0.002s] 0 tests run: 0 passed, 199 skipped
```

There is something off here...

Reply:
I've just copy and pasted the line that works for me:
`cargo nextest run -F shared-memory -F transport_shm -p zenoh-transport`
It is the same as you tried. Maybe the problem in nextest version? I use recent 0.9.57

Reply:
`dmitriy@dmitriy-TITAN:~/zenoh$ cargo nextest run -F shared-memory -F transport_shm -p zenoh-transport
warning: constant `MSG_SIZE_ALL` is never used
   --> io/zenoh-transport/tests/unicast_transport.rs:223:7
    |
223 | const MSG_SIZE_ALL: [usize; 2] = [1_024, 131_072];
    |       ^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default

warning: `zenoh-transport` (test "unicast_transport") generated 1 warning
    Finished test [unoptimized + debuginfo] target(s) in 0.11s
    Starting 12 tests across 13 binaries (13 skipped)
        PASS [   0.004s] zenoh-transport common::seq_num::tests::sn_gap
        PASS [   0.004s] zenoh-transport common::batch::tests::serialization_batch
        PASS [   0.004s] zenoh-transport common::seq_num::tests::sn_precedence
        PASS [   0.004s] zenoh-transport common::seq_num::tests::sn_generation
        PASS [   0.004s] zenoh-transport common::seq_num::tests::sn_set
        PASS [   0.007s] zenoh-transport common::pipeline::tests::tx_pipeline_blocking
        PASS [   0.087s] zenoh-transport unicast::establishment::cookie::tests::codec_cookie
        PASS [   0.162s] zenoh-transport common::pipeline::tests::tx_pipeline_flow
        PASS [   2.101s] zenoh-transport::endpoints endpoint_shm
        PASS [   5.344s] zenoh-transport::unicast_shm tests::transport_shm_shm
        PASS [   6.427s] zenoh-transport::unicast_transport transport_unicast_shm_only
        PASS [   8.378s] zenoh-transport::unicast_transport transport_unicast_shm_only_with_shm
------------
     Summary [   8.381s] 12 tests run: 12 passed, 13 skipped
dmitriy@dmitriy-TITAN:~/zenoh$ 
`

Reply:
```
dmitriy@dmitriy-TITAN:~/zenoh$ cargo nextest run -F shared-memory -F transport_shm -p zenoh-transport
warning: constant `MSG_SIZE_ALL` is never used
   --> io/zenoh-transport/tests/unicast_transport.rs:223:7
    |
223 | const MSG_SIZE_ALL: [usize; 2] = [1_024, 131_072];
    |       ^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default

warning: `zenoh-transport` (test "unicast_transport") generated 1 warning
    Finished test [unoptimized + debuginfo] target(s) in 0.11s
    Starting 12 tests across 13 binaries (13 skipped)
        PASS [   0.004s] zenoh-transport common::seq_num::tests::sn_gap
        PASS [   0.004s] zenoh-transport common::batch::tests::serialization_batch
        PASS [   0.004s] zenoh-transport common::seq_num::tests::sn_precedence
        PASS [   0.004s] zenoh-transport common::seq_num::tests::sn_generation
        PASS [   0.004s] zenoh-transport common::seq_num::tests::sn_set
        PASS [   0.007s] zenoh-transport common::pipeline::tests::tx_pipeline_blocking
        PASS [   0.087s] zenoh-transport unicast::establishment::cookie::tests::codec_cookie
        PASS [   0.162s] zenoh-transport common::pipeline::tests::tx_pipeline_flow
        PASS [   2.101s] zenoh-transport::endpoints endpoint_shm
        PASS [   5.344s] zenoh-transport::unicast_shm tests::transport_shm_shm
        PASS [   6.427s] zenoh-transport::unicast_transport transport_unicast_shm_only
        PASS [   8.378s] zenoh-transport::unicast_transport transport_unicast_shm_only_with_shm
------------
     Summary [   8.381s] 12 tests run: 12 passed, 13 skipped
dmitriy@dmitriy-TITAN:~/zenoh$ 

```

Reply:
Right, I was still with `nextest 0.9.8`. I've updated it to `0.9.57` and I get the same result as you.